plugin/pkg/utils/miscellaneous.go

```diff
@@ -35,3 +37,17 @@ func GetProject(shoot *garden.Shoot, projectLister gardenlisters.ProjectLister)
 	}
 	return nil, fmt.Errorf("no project for shoot namespace %q", shoot.Namespace)
 }
+
+// GetPlantProject returns the Project resource the passed plant is assigned to.
+func GetPlantProject(plant *gardenercore.Plant, projectLister gardenlisters.ProjectLister) (*garden.Project, error) {
+	projects, err := projectLister.List(labels.Everything())
```

Comment:
more or less exactly the same code like above.
Again, please create a separate function that takes the namespace as a parameter and simply re-use it where needed. No need to have two functions `GetShootProject` and `GetPlantProject`, instead only one `GetProjectForNamespace`.